src/include/distributed/distributed_transaction_management.h

```diff
@@ -0,0 +1,52 @@
+/*
+ * distributed_transaction_management.h
+ *
+ * Copyright (c) 2016, Citus Data, Inc.
+ *
+ *-------------------------------------------------------------------------
+ */
+
+#ifndef DISTRIBUTED_TRANSACTION_MANAGEMENT_H_
+#define DISTRIBUTED_TRANSACTION_MANAGEMENT_H_
+
+
+#include "datatype/timestamp.h"
+#include "nodes/pg_list.h"
+
+
+/*
+ * Citus identifies a distributed transaction with a triplet consisting of
+ *
+ *  -  initiatorNodeIdentifier: A unique identifier of the node that initiated
+ *     the distributed transaction
+ *  -  transactionId: A locally unique identifier assigned for the distributed
+ *     transaction on the node that initiated the distributed transaction
+ *  -  timestamp: The current timestamp of distributed transaction initiation
+ *
+ */
+typedef struct DistributedTransactionId
+{
+	uint64 initiatorNodeIdentifier;
+	uint64 transactionId;
+	TimestampTz timestamp;
+} DistributedTransactionId;
+
+
+/*
+ * Each backend's active distributed transaction information is tracked via
+ * DistributedTransactionBackendData on the shared memory.
+ */
+typedef struct DistributedTransactionBackendData
+{
+	Oid databaseId;
+	DistributedTransactionId transactionId;
+} DistributedTransactionBackendData;
+
+
```

Comment:
> I couldn't get this request, so I'll wait to change this until I fully understand what you suggest.

I don't think the struct should a) have transaction in the name b) be in a file with transaction in the name. We're going to add more per-backend data, so we shouldn't tie this to transactions.

> Also, I was excepting that the error message goes to DistributedTransactionShmemData struct?

Hm? How would that work - there can be multiple transactions being canceled at the same time...